bin/ghcrawler.js

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const start = require('../src').start;
+const path = require('path');
+
+const args = require('argly').createParser({
+    '--config -c *': 'string'
+  })
+  .parse();
+
+Promise.resolve()
```

Comment:
I was going to switch from Q to native Promises everywhere, but realized you guys were using some more obscure features of Q that required some more thought. We can switch that to Q, but I'm not sure it really matters since `Promise` implementations are all compatible as like as the objects are "Promise-like".